File: compass/Models/NightPlanner.cs
using MongoDB.Bson.Serialization.Attributes;

namespace compass.Models;

[BsonIgnoreExtraElements]
public class NightPlanner : Entity
{
    [BsonElement("title")]
    public string Title { get; set; }
    
    [BsonElement("description")]
    public string Description { get; set; }
    
    [BsonElement("locationCityId")]
    public int LocationCityId { get; set; }

    [BsonElement("locationCity")]
    public string LocationCity { get; set; }
    
    [BsonElement("locationplans")]
    public List<LocationPlan> LocationPlans { get; set; } 
}

[BsonIgnoreExtraElements]
public class LocationPlan : Entity
{
    [BsonElement("locationdata_id")]
    public string locationdata_id { get; set; }
    
    [BsonElement("duration")]
    public int Duration { get; set; }
    
    [BsonElement("order")]
    public int Order { get; set; }
    
    [BsonElement("longitude")]
    public float? Longitude { get; set; }
    
    [BsonElement("latitude")]
    public float? Latitude { get; set; }
}

File: server/src/models/NightPlanner.js
const mongoose = require('mongoose');

const locationPlanSchema = new mongoose.Schema({
  locationdata_id: { type: String, required: true },
  duration: { type: Number, required: true },
  order: { type: Number, required: true },
  longitude: { type: Number, required: true },
  latitude: { type: Number, required: true },
});

const nightPlannerSchema = new mongoose.Schema({
  title: { type: String, required: true },
  description: { type: String, required: true },
  locationCityId: { type: Number, required: true },
  locationCity: { type: String, required: true },
  locationPlans: { type: [locationPlanSchema], required: true },
});

const NightPlannerModel = mongoose.model('NightPlanner', nightPlannerSchema);

module.exports = NightPlannerModel;
